refactored the palette.js code

fakela opened the pull request.

walterbender commented:
Great that is it loading correctly now.
Two questions:
(1) Cannot we use the js files in js/ instead of copying them into dist/???
(2) I don't understand what the promisetest is doing. Can you explain it?

fakela commented:
Alright would change the file directory to js, I wanted to use dist so I
don't have confusion at first and also I will explain what the promise test
is doing , I wanted to  write an asynchronous test using Jasmine with no
errors

On Sun, Jun 23, 2019, 11:23 PM Walter Bender <notifications@github.com>
wrote:

> Great that is it loading correctly now.
> Two questions:
> (1) Cannot we use the js files in js/ instead of copying them into dist/???
> (2) I don't understand what the promisetest is doing. Can you explain it?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/sugarlabs/musicblocks/pull/1793?email_source=notifications&email_token=AJL5DA4UYSWSQIERSYPKWF3P37ZWBA5CNFSM4H2VBOH2YY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGODYLIOLY#issuecomment-504792879>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AJL5DA2EGI4A3PF4EXYBDQ3P37ZWBANCNFSM4H2VBOHQ>
> .
>

walterbender commented:
@fakela Maybe try: js/palette.js instead of js/*js ???

walterbender commented:
Or add the missing library calls as well?
lib/*js

fakela commented:
Lemme try js/palette.js

On Mon, Jun 24, 2019, 3:34 PM Walter Bender <notifications@github.com>
wrote:

> Or add the missing library calls as well?
> lib/*js
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/sugarlabs/musicblocks/pull/1793?email_source=notifications&email_token=AJL5DA5VCSPWUN7G2BLICD3P4DLPDA5CNFSM4H2VBOH2YY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGODYNECQI#issuecomment-505037121>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AJL5DA4MTUPQ7Y3CIGENCR3P4DLPDANCNFSM4H2VBOHQ>
> .
>

fakela commented:
Have fixed the file path error

walterbender commented:
Great that the config is working. Can you please explain how tests/palette.js actually tests the palette code?

fakela commented:
I renamed the test/palette.js cause am testing a function in isolation . The flow of the test is
I) set completed to false
ii) create promise
iii)Run promise callback which sets completed to true
iv) Expect that completed is true 
i used the async/ await as this relieves you from putting asserts in the then and catch block

fakela commented:
karma wasnt seeing the other test so i had to change the config file to test/ *js it failed though

walterbender commented:
I am wondering about the namespace for the methods used in the tests. You use delayExecution() from palette.js in the promise-palette test and delay() from blocks.hjs in the promise-blocks test. Maybe something like:

delayPalette()
delayBlocks()

??? 

I still don't quite understand how these tests work... I'll read up.

fakela commented:
the links to the test   
https://jsperf.com/threesss between the code i added before and the refactored code
https://jsperf.com/newss/1 between the old code and the code i added
